src/fsharp/NameResolution.fs

```diff
@@ -1538,14 +1538,18 @@ type TcResultsSinkImpl(g, ?source: string) =
     let formatStringCheckContext =
         lazy
             source |> Option.map (fun source ->
-                let source = source.Replace("\r\n", "\n").Replace("\r", "\n")
                 let positions =
-                    source.Split('\n')
-                    |> Seq.map (fun s -> String.length s + 1)
-                    |> Seq.scan (+) 0
-                    |> Seq.toArray
-                { NormalizedSource = source 
-                  LineEndPositions = positions })
+                    [|
+                        yield 0
+                        for i in 0..source.Length-1 do
+                            let c = source.[i]
+                            if c = '\r' && i + 1 < source.Length && source.[i+1] = '\n' then ()
```

Comment:
style/readbility comment: looping through `1..source.Length` instead of `0..source.Length-1` would make more sense here, since we already processed the 0th byte with the `yield 0` above the loop. This would also avoid the `i+1` in four places:
```fs
 [|
    yield 0
    for i in 1..source.Length do
         let c = source.[i-1]
         if c = '\r' && i < source.Length && source.[i] = '\n' then ()
         elif c = '\r' then yield i
         elif c = '\n' then yield i
    yield source.Length
 |]
```

Reply:
yeah I think we should avoid duplicates as well.